skfp/distances/all_bit.py

```diff
@@ -0,0 +1,365 @@
+        return 1.0
+
+    if isinstance(vec_a, csr_array) and isinstance(vec_b, csr_array):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+        return _all_bit_binary_scipy(vec_a_bool, vec_b_bool)
+
+    elif isinstance(vec_a, np.ndarray) and isinstance(vec_b, np.ndarray):
+        vec_a_bool = vec_a.astype(bool)
+        vec_b_bool = vec_b.astype(bool)
+        return _all_bit_binary_numpy(vec_a_bool, vec_b_bool)
+
+    else:
+        raise TypeError(
+            f"Both vec_a and vec_b must be of the same type, either numpy.ndarray "
+            f"or scipy.sparse.csr_array, got {type(vec_a)} and {type(vec_b)}"
+        )
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def all_bit_binary_distance(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    All-Bit distance for vectors of binary values.
+
+    Computes the All-Bit distance for binary data between two input arrays
+    or sparse matrices by subtracting the similarity from 1, using to
+    the formula:
+
+    .. math::
+
+        dist(vec_a, vec_b) = 1 - sim(vec_a, vec_b)
+
+    The calculated distance falls within the range ``[0, 1]``.
+    Passing all-zero vectors to this function results in a distance of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    -------
+    distance : float
+        All-Bit distance between ``vec_a`` and ``vec_b``.
+
+    Examples
+    --------
+    >>> from skfp.distances import all_bit_binary_distance
+    >>> import numpy as np
+    >>> vec_a = np.array([1, 0, 1])
+    >>> vec_b = np.array([1, 0, 1])
+    >>> dist = all_bit_binary_distance(vec_a, vec_b)
+    >>> dist  # doctest: +SKIP
+    0.0
+
+    >>> from skfp.distances import all_bit_binary_distance
+    >>> from scipy.sparse import csr_array
+    >>> vec_a = csr_array([[1, 0, 1]])
+    >>> vec_b = csr_array([[1, 0, 1]])
+    >>> dist = all_bit_binary_distance(vec_a, vec_b)
+    >>> dist  # doctest: +SKIP
+    0.0
+    """
+    return 1 - all_bit_binary_similarity(vec_a, vec_b)
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def all_bit_count_similarity(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    r"""
+    All-bit similarity for vectors of count values.
+
+    Computes the all-bit for count data between two input arrays
+    or sparse matrices using the formula:
+
+    .. math::
+
+        sim(vec_a, vec_b) = \frac{\text{count}_a - \text{count}_\text{diff}}{\text{count}_a}
+
+    where:
+
```

Comment:
This similarity does not work for count version due to division by `n`, and we shouldn't modify that. So remove the count variants.